src/Bowling.js

```diff
@@ -0,0 +1,112 @@
+var Bowling = function(){
+  this.rollsArray = [];
+  this.scoresArray = [];
+  this.rollCounter = 0;
+  this.frameCounter = 1;
+};
+
+Bowling.prototype.roll = function(rollScore){
+  var self = this;
+
+  function throwAlert(){
+    alert("Illegal score");
+  };
+
+  function isIllegalScore(){
+    return self.frameCounter % 2 === 0 && (rollScore + self.scoresArray[self.scoresArray.length - 1] > 10);
+  };
+
+  if (isIllegalScore()){
+    throwAlert();
+  } else {
+    self.rollsArray.push(rollScore);
+    self.registerRoll(rollScore);
+  };
+};
+
+Bowling.prototype.registerRoll = function(rollScore){
+  var self = this;
+
+  function tenthFrame (){
+    return self.frameCounter > 18;
+  };
+
+  function isStrike(){
+    return rollScore === 10;
+  };
+
+  function isFrameEnd(){
+    return self.frameCounter % 2 === 0;
+  };
+
+  function isSpare(){
+    return rollScore + self.scoresArray[self.scoresArray.length - 1] === 10;
+  };
+
+  function decideSpare(){
+    if (isSpare()){
+      self.scoresArray.push('/');
+    } else {
+      self.scoresArray.push(rollScore);
+    };
+    self.frameCounter ++;
+  };
+
+  function pushScore(){
+    self.scoresArray.push(rollScore);
+  };
+
+  if (tenthFrame()){
+    pushScore();
+  } else if (isStrike()){
+    pushScore();
+    self.scoresArray.push('X');
+    self.frameCounter += 2;
+  } else if (isFrameEnd()){
+    decideSpare();
+  } else {
+    pushScore();
+    self.frameCounter ++;
+  };
+  self.rollCounter ++;
+};
```

Comment:
Have the helper functions below the main execution - helps a lot for readability